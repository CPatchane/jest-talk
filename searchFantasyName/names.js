export default [
  'Lafchris',
  'Richcome',
  'Markke',
  'Retferth',
  'Nathniel',
  'Laydun',
  'Sigcuth',
  'Thy',
  'Ceolliam',
  'Ard',
  'Cwenkev',
  'Palau',
  'Thasrichham',
  'Misere',
  'Dinoke',
  'Ruthbar',
  'Thochael',
  'Muelsig',
  'Cas',
  'Eadonbri',
  'Amarr',
  'Etuari',
  'Fresdam',
  'Naldbert',
  'Gard',
  'Mondan',
  'Georna',
  'Wardlaf',
  'Ricro'
]
