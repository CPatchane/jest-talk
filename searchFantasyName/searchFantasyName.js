// searchFantasyName.js

import names from './names'

function searchFantasyName (query) {
  return names.filter(
    name => name.search(
        new RegExp(query, 'i')
      ) !== -1
  )
}

export default searchFantasyName
