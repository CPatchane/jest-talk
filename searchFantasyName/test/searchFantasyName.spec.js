// searchFantasyName.spec.js
/* eslint-env jest  */

import searchFantasyName from '../searchFantasyName'

describe('searchFantasyName', () => {
  it('should return results from "am" query', () => {
    expect(searchFantasyName('am')).toMatchSnapshot()
  })
})
