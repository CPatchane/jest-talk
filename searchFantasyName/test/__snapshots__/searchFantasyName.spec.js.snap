// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchFantasyName should return results from "am" query 1`] = `
Array [
  "Ceolliam",
  "Thasrichham",
]
`;
